wdl-format/src/element.rs

```diff
@@ -0,0 +1,508 @@
+//! Elements used during formatting.
+
+use std::collections::HashMap;
+use std::iter::Peekable;
+
+use nonempty::NonEmpty;
+use wdl_ast::AstToken as _;
+use wdl_ast::Element;
+use wdl_ast::Node;
+use wdl_ast::SyntaxKind;
+
+use crate::NEWLINE;
+use crate::PreToken;
+use crate::TokenStream;
+use crate::Writable;
+
+pub mod node;
+
+/// Trivia associated with some more formidable element.
+///
+/// Trivia would be things like comments and whitespace.
+#[derive(Clone, Debug, Default)]
+pub struct Trivia {
+    /// Any preceeding trivia.
+    preceeding: Option<NonEmpty<Box<FormatElement>>>,
+
+    /// Any inline trivia.
+    inline: Option<NonEmpty<Box<FormatElement>>>,
+}
+
+impl Trivia {
+    /// Any preceeding trivia that are not whitespaces.
```

Comment:
```suggestion
    /// Any preceding trivia that are not whitespaces.
```
There are situations where we want to support optional blank lines, and those are going to require us to be mindful of preceding whitespace. Inline whitespace can always be ignored though.